crates/networking/p2p/discv4/server.rs

```diff
@@ -488,13 +488,12 @@ impl DiscoveryServer {
     ) -> Result<(), DiscoveryServerError> {
         let node_id = node_id(&sender_public_key);
 
-        let contact = self
-            .validate_contact(sender_public_key, node_id, from, "ENRResponse")
-            .await?;
-
-        if !contact.has_pending_enr_request() {
-            debug!(received = "ENRResponse", from = %format!("{sender_public_key:#x}"), "unsolicited message received, skipping");
-            return Err(DiscoveryServerError::InvalidContact);
+        if self
+            .validate_enr_response(sender_public_key, node_id, from)
+            .await
+            .is_err()
+        {
+            return Ok(());
```

Comment:
This function (`validate_enr_response`) returns a Contact element. Given that this call is not using the return value, I think we could have a version returning unit type (i.e. Result of nothing).